### ISSUE TITLE
OS Error: Operation not permitted

### ISSUE THREAD

**Issue by long1eu** [renefloor/flutter_cached_network_image#8]
I'm getting : Another exception was thrown: FileSystemException: Creation failed, path = '/var/mobile/Containers/Data/Application/984BEEDA-D355-4429-AD4C-D6D211082B7A' (OS Error: Operation not permitted, errno = 1)
on iOS. Any idea?

**Comment by renefloor** [comment]
Hmm not sure. I have some questions so I can try to replicate it.
Library version
iOS version
Url you try to load.

If you use 0.2.0, is this an image that also gave problems in 0.1.0? I guess iOS doesn't like files without an extension.

**Comment by long1eu** [comment]
version 0.2.0
iOS 11.2.1
https://res.cloudinary.com/du8gpvdbl/image/upload/v1511088243/0-1511088243104.png but I'm getting the error for every url

**Comment by renefloor** [comment]
Hmm raar. Ik ga het uitzoeken.

**Comment by long1eu** [comment]
Bedankt!

**Comment by renefloor** [comment]
I cannot reproduce it, but my path is the following:
/Users/rene/Library/Developer/CoreSimulator/Devices/E480FE42-3C1C-477D-AA2F-B5B74FF244FB/data/Containers/Data/Application/31970312-CC93-4FDB-9F87-7EE9176B641D/Library/Caches/cache/601ee310-d2d0-11e7-b450-2b33870382f5

So after Application/{id} it still has Library/Caches/cache/...
I can imagine that if it tries to create the folder for the application it fails.

**Comment by long1eu** [comment]
I'll take a look deeper, thanks

**Comment by polbatllo** [comment]
I'm also having this issue, although it's only happening in a real device, not in the simulator.

**Comment by renefloor** [comment]
Hmm.. I am mostly an Android dev, so I have no experience at all with the iOS file system. But I am trying to get my head around it.
I tried it on an iPad and that worked fine. Do you have a release build or do you build from Intellij? Do you try it on an iPad or an iPhone.

**Comment by polbatllo** [comment]
Not at the minute.. but to reproduce you need to:
- On a physical device, run the application, load the images you want and then, shut down the app, completely.
- Run the application again and head to those images.

Once I have a computer around I'll gather some logs and attach them here :)

**Comment by renefloor** [comment]
I just had a look at the [apple docs on the file system](https://developer.apple.com/library/content/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW28).
My current suspection is that the part before /Library is not fixed. I'll have a look at that when I have access to an iOS device ;).
If you are earlier than me, try to log `getTemporaryDirectory()` and then log the path of the file.

Maybe it would be better to only store the filename anyhow and get the temp directory every time the cache manager is initialized.

**Comment by renefloor** [comment]
Could you confirm that the issue is fixed if you use the following dependency?
```
cached_network_image:
    git:
      url: git://github.com/renefloor/flutter_cached_network_image.git
      ref: develop
```

**Comment by polbatllo** [comment]
Sorry for the delay @renefloor . I've been on holiday. I confirm the issue is fixed on that dependency.

Awesome.

**Comment by renefloor** [comment]
No problem, I'll probably make a new release today.

**Comment by renefloor** [comment]
Just released the fix in 0.2.1

**Comment by polbatllo** [comment]
Awesome. Thanks.
